rio/src/org/usfirst/frc/team5806/robot/CameraClient.java

```diff
@@ -0,0 +1,141 @@
+import java.net.*;
+import java.io.*;
+import java.awt.*;
+import java.awt.image.*;
+import java.util.Arrays;
+import javax.imageio.ImageIO;
+import javax.swing.*;
+
+public class CameraClient {
+    private String hostIP;
+    private int serverPort;
+    private Socket client;
+    private OutputStream out;
+    private InputStream in;
+    
+    public CameraClient(String hostIP, int serverPort) {
+        this.hostIP = hostIP;
+        this.serverPort = serverPort;
+    }
+    public CameraClient() {}
+    public void sethostIP(String hostIP) {
+        this.hostIP = hostIP;
+    }
+    public void setServerPort(int port) {
+        this.serverPort = port;
+    }
+    public String getHostIP() {
+        return hostIP;
+    }
+    public int getServerPort() {
+        return serverPort;
+    }
+    public void initSocket() {
+        try {
+            this.client = new Socket(hostIP, serverPort);
+            this.out = this.client.getOutputStream();
+            this.in = this.client.getInputStream();
+            System.out.println("Client socket successfully initialized.");
+        } catch (IOException e) {
+            System.out.println("Error Initializing Socket or Input/Output Streams");
+        }
+        
+    }
+    public void writeString(String s) {
+        try {
+            BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(out));
+            bw.write(s);
+            bw.close();
+        } catch (IOException e) {
+            System.out.println("Error writing string to socket.");
+        }
+    }
+    public String readSocketString() {
+        String s = null;
+        try {
+            BufferedReader br = new BufferedReader(new InputStreamReader(in));
+            s = br.readLine();
+            br.close();
+        } catch (IOException e) {
+            System.out.println("Error reading string from socket.");
+        }
+        return s;
+    }
+    private byte[] readSocketBytes() {
+        ByteArrayOutputStream bs = null;
+        try {   
+            bs = new ByteArrayOutputStream();
+            byte[] buffer = new byte[10000];
+            int s = 0;
+            while ((s = in.read(buffer, 0, 10000)) != -1) {
+                if ((char)buffer[s-1] == 'e'){
```

Comment:
Believe you're seeing your desync here. If you get more image data than your buffer (which is totally OK, you can just do more buffer reads), then you're gonna miss the `e`. This is b/c you're checking for the `e` at the end of your buffer read when it could be somewhere in the middle.